### PR TITLE
Overhaul Overview Docs

### PR DESCRIPTION
**Pull request by LordGhostX**
TL;DR of what happened:

- Merged the introduction, motivation, and how it works pages into a comprehensive "What is Waku?" guide
- Added an image from [waku.org/about](https://waku.org/about#history) to the history page
- Updated guides to include suggestions made to the ETHTaipei and ETHDenver slides
- Added a dedicated "Use Cases" page
- Reworded content throughout multiple pages
- Updated the Docusaurus version, footer links, and README file to provide expanded installation steps

**Please note that this PR is still in draft status; therefore, please refrain from merging it. However, you are welcome to review and provide suggestions.**

Next steps include creating a dedicated landing page for the documentation, overhauling the concept docs, and addressing [issue #10](https://github.com/waku-org/waku.guide/issues/10).

### CODE REVIEW

File: docs/overview/implementations.md [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+The JavaScript/TypeScript implementation of the Waku protocol designed for use in browser environments.
 
-#### [go-waku](https://github.com/status-im/go-waku)
+### [go-waku](https://github.com/waku-org/go-waku)
 
-A Go implementation for native integration in go applications, C-Bindings are also available with C#, Swift and Kotlin examples.
+The Golang implementation of the Waku protocol designed for native integration in Golang applications, includes C bindings with examples available in C#, Swift, and Kotlin.
 
-#### [@waku/react-native](https://github.com/waku-org/waku-react-native)
+### [@waku/react-native](https://github.com/waku-org/waku-react-native)
```
Comment (fryorcraken) [inline docs/overview/implementations.md:19]: Can also mention the Rust bindings: github.com/waku-org/waku-rust-bindings

File: docs/overview/use-cases.md
Comment (fryorcraken) [inline docs/overview/use-cases.md:1]: this page goes a bit deeper in the use cases, it might be of interest: https://js.waku.guide/docs/use_cases/

File: docs/overview/what-is-waku.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+---
+title: What is Waku?
+slug: /
+---
+
+Waku is a family of peer-to-peer protocols that provide **secure and private communication** in a decentralized environment. It is designed to operate in **resource-limited environments**, making it suitable for various decentralized applications (dApps).
```
Comment (fryorcraken) [inline docs/overview/what-is-waku.md:6]: The phrasing might make a developer think that Waku is **only** for resource-limited environments. Maybe worth noting it can also be used as a node or desktop application.

File: docs/overview/what-is-waku.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+Many applications require some form of messaging protocol to facilitate communication between different subsystems or nodes. Waku is flexible and can be used for human-to-human and machine-to-machine messaging scenarios.
+
+### Peer-to-Peer
+
+Waku is suitable for applications that require a peer-to-peer approach, as it offers the following advantages:
+
+- Censorship resistance with no single point of failure.
+- Adaptive and scalable network.
+- Shared infrastructure/service network.
```
Comment (fryorcraken) [inline docs/overview/what-is-waku.md:24]: As discussed in our call, might be worth presenting the concept of Waku as a service network

File: docs/overview/what-is-waku.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+Waku protocols ensure that users communication remains **censorship-resistant and privacy-preserving**, giving them complete control over their data. By incorporating Waku into your dApp, you can add decentralized communication features to your application **without compromising security or privacy**.
+
+## Motivation and Goals
+
+The Waku family of protocols is designed for diverse applications due to their properties, such as:
+
+### Generalized Messaging
+
+Many applications require some form of messaging protocol to facilitate communication between different subsystems or nodes. Waku is flexible and can be used for human-to-human and machine-to-machine messaging scenarios.
```
Comment (fryorcraken) [inline docs/overview/what-is-waku.md:16]: Something worth mentioning that Waku aims to solve the problem of _ephemeral_ messaging. Waku does not intend to be a data storage system.

File: docs/overview/what-is-waku.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+- Pseudonymity and not being tied to any personally identifiable information (PII).
+- Metadata protection in transit.
+- Various forms of unlinkability.
+
+### Modular Design
+
+Waku nodes are adaptive and can be customized based on the application's requirements and environment. Users can adjust several parameters, including:
+
+- Low privacy/low resource usage vs. high privacy/increased latency + bandwidth usage.
```
Comment (fryorcraken) [inline docs/overview/what-is-waku.md:42]: may be worth mentioning this is the anonymity trilemma https://eprint.iacr.org/2017/954.pdf

File: docs/overview/why-waku.md [hunk 6]
```diff
@@ -2,31 +2,23 @@
...
+Waku is intended to empower individuals by returning control of communication to them. It is the go-to standard for Web3 communication, offering a **scalable decentralized communication** solution.
+
+- Waku improves upon Whisper's capabilities by overcoming limitations and addressing functional gaps.
+- It provides a public infrastructure for the Ethereum and multi-chain ecosystem, serving as a common good.
+- It is not confined to a particular blockchain.
+- It is modular, adaptable, and can cater to various use cases.
+- It allows developers to decentralize communication in their dApps or move actions off-chain while maintaining decentralization.
+- It can run on various platforms, including mobile devices, cloud environments, web browsers, native desktop apps, or even dapp nodes!
+
+## Whisper vs. Waku: Why Waku Wins
```
Comment (fryorcraken) [inline docs/overview/why-waku.md:16]:
```suggestion
## Whisper vs. Waku: Why Waku is Necessary
```
Just an idea as it's not us vs them but more a "whisper is not good enough and was abandonned"

File: docs/overview/why-waku.md [hunk 7]
```diff
@@ -2,31 +2,23 @@
...
+
+| | Whisper | Waku |
+| - | - | - |
+| **Scalability** | Whisper doesn't scale very well, specifically when it comes to bandwidth usage on mobile devices. | Uses sharding to partition the network into separate gossipsub networks, while maintaining a shared discovery network for peer location. |
```
Comment (fryorcraken) [inline docs/overview/why-waku.md:20]: @jm-clius @alrevuelta as recently discussed, while we accept that for Status Communities, sharding is necessary, we don't want to push it to all platforms.
Shall we keep sharding out of the introductory doc then? and just mentioned floodsub vs gossipsub?

**Review by fryorcraken — COMMENTED** [review]
First pass of feedback. Great work.

IF it's still "draft" you can mark the PR as "draft" in GitHub, it'll also prevent any merging.

File: docs/overview/why-waku.md (hunk 7 above)
Comment (jm-clius) [reply docs/overview/why-waku.md:20]: I think so. In fact, we aim to introduce automatic sharding so that developers wouldn't have to think about it at all. Of course, the technical explanation of how this works will be of interest to some, but this would not be introductory material.

**Review by hackyguru — APPROVED** [review]

**Comment by LordGhostX** [comment]
@hackyguru I already renamed the heading to "Getting Started" and the "What is Waku?" pages lives inside that

Screenshot 2023-05-10 at 20 03 02

I also plan to have a dedicated "homepage" which will serve as a landing page and link to various parts of the docs: Getting Started, js-waku, nwaku, etc

File: docusaurus.config.js [hunk 8]
```diff
@@ -67,16 +63,28 @@ const config = {
           },
         ],
       },
-        announcementBar: {
-        id: "support_us",
+      announcementBar: {
         content:
-          "Waku is not production ready yet. Join our <a href='https://discord.gg/Nrac59MfSX'>discord community</a> to stay updated.",
+          "While Waku still has progress to make, apps like <a href='/overview/who-is-using-waku'>XMTP and Railgun</a> have already implemented it in production. Join our <a href='https://discord.gg/Nrac59MfSX'>Discord Community</a> to stay updated on our progress.",
```
Comment (fryorcraken) [inline docusaurus.config.js:68]:
```suggestion
          "While Waku still has progress to make, apps like <a href='/overview/who-is-using-waku'>XMTP and Railgun</a> have already implemented it in production. Join our <a href='https://discord.gg/Nrac59MfSX'>Discord Community</a> to stay updated on our progress.",
```
Let's adjust the message again, needs to be a bit stronger. Maybe something on the line "Using Waku comes with a number of risks and limitations..."

**Review by fryorcraken — APPROVED** [review]
